articles/101_build_tool.md

```diff
@@ -0,0 +1,212 @@
+
+The tool only supports CMake-based packages and builds each package in topological order using the command sequence common for CMake packages: `cmake`, `make`, `make install`.
+While each package can parallelize the build of its targets the packages are processed sequentially even if they are not (recursive) dependencies of each other.
+
+### ament_tools
+
+`ament_tools` is provided by a standalone Python 3 package used to build ROS 2 packages.
+It was developed to bootstrap the ROS 2 project, is therefore only targeting Python 3, and works on Linux, MacOS and Windows.
+In addition to CMake packages it also supports building Python packages and can infer meta information without requiring an explicit package manifest.
+The tool performs an "isolated" build like `catkin_make_isolated` (one CMake invocation per package) but also parallelizes the build of packages which have no (recursive) dependencies on each other.
+
+### catkin_tools
+
+[catkin_tools](https://catkin-tools.readthedocs.io/) is provided by a standalone Python package used to build ROS 1 packages.
+It was developed after `catkin_make` / `catkin_make_isolated` to build packages in parallel as well as provide significant usability improvements.
+The tool supports building CMake packages and builds them in isolation as well as support parallelizing the process across packages.
+
+### Naming
+
+The existing build tools are all named by the build system they are supporting.
+In general it should be possible for a build tool to support multiple different build systems.
+Therefore a name for a build tool being derive from a single build system might mislead the users that the tool only works for that specific build system.
+To avoid confusion of the user it would be beneficial if the build tool would have a different name to avoid implying an undesired correlation.
+
+## Rational
+
+Currently each of the existing tools performs similar tasks and duplicates a significant amount of the logic.
+Certain features are only available in some of the tools while other tools lack those.
+
+The reason to work on a single universal build tool comes down to reduce the effort necessary for development and maintenance.
+Additionally this makes new features developed for one ROS version / build system available to the other supported ROS versions / build systems.
+
+## Requirements
+
+The unified build tool should provide a superset of the functionality provided by the existing tools.
+In the following a few use cases are described as well as desired software criteria.
+
+### Use Cases
+
+The following uses cases should be satisfied by the unified build tool.
+
+#### Build ROS 1 workspaces
+
+The tool needs to be able to build ROS 1 workspaces which can already be built using `catkin_make_isolated` / `catkin_tools`.
+
+#### Build ROS 2 workspaces
+
+The tool needs to be able to build ROS 2 workspaces which can already be built using `ament_tools`.
+
+#### Build Gazebo including dependencies
```

Comment:
From the 'goal' section above I understood that this was just a 'nice-to-have'. It would be good to either mention again that this is a nice-to-have, or explain why it is that this is a requirement